src/compat.rs

```diff
@@ -9,21 +9,36 @@ use strum_macros::{EnumCount as EnumCountMacro, EnumIter};
 
 /// Version of the Landlock [ABI](https://en.wikipedia.org/wiki/Application_binary_interface).
 ///
-/// `ABI` enables to get the features supported by a specific Landlock ABI.
+/// `ABI` enables to get the features supported by a specific Landlock ABI
+/// (without relying on the kernel version which may not be accessible or patched).
 /// For example, [`AccessFs::from_all(ABI::V1)`](Access::from_all)
 /// gets all the file system access rights defined by the first version.
 ///
 /// Without `ABI`, it would be hazardous to rely on the the full set of access flags
 /// (e.g., `BitFlags::<AccessFs>::all()` or `BitFlags::ALL`),
 /// a moving target that would change the semantics of your Landlock rule
-/// when migrating to a newer version of this crate
-/// (i.e. non-breaking change with new supported features).
-/// This usage should then be considered indeterministic because requested features
-/// (e.g., access rights)
-/// could not be tied to the application source code.
+/// when migrating to a newer version of this crate.
+/// Indeed, a simple `cargo update` or `cargo install` performed by any developer
+/// can result in a new version of this crate (fixing bugs or bringing non-breaking changes).
+/// This crate cannot give any guarantee concerning the new restrictions resulting from
+/// these unknown bits (i.e. access rights) that would not be controlled by your application but by
+/// a future version of this crate instead.
+/// Because we cannot know what would be the effect of an unknown restriction on your application,
+/// handling an untested Landlock access right (i.e. denied-by-default access)
```

Comment:
Probably "when handling an untested..." is better

Reply:
I'm not sure what you mean here, but I updated the doc. Please let me know what you think.

Reply:
I had envisioned keeping the order as is and just adding the word "when", ie "Because we cannot know what the effect of an unknown restriction on your application would be when handling an untested landlock access right...".

The way you updated it is fine, but feels a little awkward stylistically to me.